gmso/external/convert_hoomd.py

```diff
@@ -0,0 +1,1498 @@
+        snapshot.impropers.typeid[0:] = improper_typeids
+        snapshot.impropers.group[0:] = np.reshape(improper_groups, (-1, 4))
+    elif isinstance(snapshot, gsd.hoomd.Snapshot):
+        snapshot.impropers.types = unique_improper_types
+        snapshot.impropers.typeid = improper_typeids
+        snapshot.impropers.group = np.reshape(improper_groups, (-1, 4))
+
+    warnings.warn(f"{top.n_impropers} impropers detected")
+    warnings.warn(
+        f"{len(unique_improper_types)} unique dihedral types detected"
+    )
+
+
+def _prepare_box_information(top):
+    """Prepare the box information for writing to gsd."""
+    lx = ly = lz = xy = xz = yz = 0.0
+    if allclose_units(
+        top.box.angles, np.array([90, 90, 90]) * u.degree, rtol=1e-5, atol=1e-8
+    ):
+        warnings.warn("Orthorhombic box detected")
+        lx, ly, lz = top.box.lengths
+        xy, xz, yz = 0.0, 0.0, 0.0
+    else:
+        warnings.warn("Non-orthorhombic box detected")
+        u_vectors = top.box.get_unit_vectors()
+        lx, ly, lz = top.box.lengths
+        xy = u_vectors[1][0]
+        xz = u_vectors[2][0]
+        yz = u_vectors[2][1]
+    return lx, ly, lz, xy, xz, yz
+
+
+def to_hoomd_forcefield(
+    top,
+    nlist_buffer=0.4,
+    r_cut=0,
+    pppm_kwargs={"resolution": (8, 8, 8), "order": 4},
+    base_units=None,
+    auto_scale=False,
+):
+    """Convert the potential portion of a typed GMSO to hoomd forces.
+
+    Parameters
+    ----------
+    top : gmso.Topology
+        The typed topology to be converted
+    nlist_buffer : float, optional, default=0.4
+        Neighborlist buffer for simulation cell. Its unit is the same as that
+        used to defined GMSO Topology Box.
+    r_cut : float
+        r_cut for the nonbonded forces.
+    pppm_kwargs : dict
+        Keyword arguments to pass to hoomd.md.long_range.make_pppm_coulomb_forces().
+    base_units : dict or str, optional, default=None
+        The dictionary of base units to be converted to. Entries restricted to
+        "energy", "length", and "mass". There is also option to used predefined
+        unit systems ("MD" or "AKMA" provided as string). If None is provided,
+        this method will perform no units conversion.
+    auto_scale : bool or dict, optional, default=False
+        Automatically scaling relevant length, energy and mass units.
+        Referenced mass unit is obtained from sites' masses.
+        Referenced energy and distance are refered from sites' atom types (when applicable).
+        If the referenced scaling values cannot be determined (e.g., when the topology is not typed),
+        all reference scaling values is set to 1.
+        A dictionary specifying the referenced scaling values may also be provided for this argument.
+
+    Returns
+    -------
+    forces : dict
+        Each entry
+    """
+    potential_types = _validate_compatibility(top)
+    base_units = _validate_base_units(
```

Comment:
It might be helpful to return base_units as well, similar to what we do in `create_hoomd_forcefield`

Reply:
`base_units` is now returned along with the forces/snapshot